### PR TITLE
Update spec for quotation mark replacement

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

Related to:
- #12552

I'm not sure why, but the pre-compiling of assets triggered Rails to render `style="..."` instead of `style='...'` in this case. But when assets are compiled on-demand, we get the single quotes. So I changed the spec to be agnostic of this detail. We actually just want to know about the link and its href.

#### What should we test?

- Specs only

#### Release notes

Changelog Category (reviewers may add a label for the release notes):

- [ ] User facing changes
- [ ] API changes (V0, V1, DFC or Webhook)
- [x] Technical changes only
- [ ] Feature toggled

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/mailers/subscription_mailer_spec.rb [hunk 1]
```diff
@@ -41,11 +41,9 @@
     end
 
     describe "linking to order page" do
-      let(:order_link_href) { "href=\"#{order_url(order)}\"" }
-      let(:order_link_style) { "style='[^']+'" }
-
       let(:shop) { create(:enterprise, allow_order_changes: true) }
 
+      let(:content) { Capybara::Node::Simple.new(body) }
```
Comment (dacook) [inline spec/mailers/subscription_mailer_spec.rb:46]: Clever! Now we can use capybara expectations just like on a web page. ✨ 

🤔 I wonder if we should build this in to a mailer spec helper...

Comment (dacook) [reply spec/mailers/subscription_mailer_spec.rb:46]: Do you know what the style check was for? I doubt it's relevant to these test expectations so I'm happy to see it go.

**Review by dacook — APPROVED** [review]
💯  Great change!

**Comment by dacook** [comment]
I'm going to merge, but FYI @rioug, we can use normal Capybara expectations in mailer specs ☝️

File: spec/mailers/subscription_mailer_spec.rb (hunk 1 above)
Comment (mkllnk) [reply spec/mailers/subscription_mailer_spec.rb:46]: I think that it was just part of the regular expression to match the link but nothing else. It wasn't checking the style.

**Pull request merged**